### PR TITLE
disable case updates for 'registry cases'

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
When using registry case search the case can not be updated directly via normal case management since they may belong to a different project space. 

This PR updates the case management UI and xform generation code to disable the case update sections. Cases can still be opened as normal.

Any updates to the cases loaded from the registry must be done indirectly using save-to-case. See https://confluence.dimagi.com/display/USH/Data+Registry#DataRegistry-Updatingcases

## Technical Summary
When action is 'followup' and the module is configured with 'registry case search':
* Disable case update UI
* Disable adding case blocks into the form to update the case 

Best reviewed by commit.

## Feature Flag
DATA_REGISTRY

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Code is well covered by tests. New specific to this workflow are added.

### QA Plan
Will be QA'd on staging

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/xform.py [hunk 1]
```diff
@@ -1567,12 +1567,13 @@ def _create_casexml_2(self, form):
...
 
+        module = form.get_module()
+        is_registry_case = _module_offers_registry_search(module) and not form_opens_case
         delegation_case_block = None
         if not actions or (form.requires == 'none' and not form_opens_case):
             case_block = None
         else:
-            case_block = XFormCaseBlock(self)
-            module = form.get_module()
+            case_block = XFormCaseBlock(self) if not is_registry_case else None
```
Comment (orangejenny) [inline corehq/apps/app_manager/xform.py:1576]: Won't setting `case_block` to `None` potentially fail later on, like the `case_block.add_create_block` in the `if form_opens_case` block below?

Comment (esoergel) [reply corehq/apps/app_manager/xform.py:1576]: I think not as it's currently written, though the logic here is kinda hard to follow.  `case_block` is always defined if `form_opens_case = True`.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/app_manager/xform.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/xform.py:1576]: ohhh, I get it now - thanks!

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/xform.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/xform.py]: I agree that this approach isn't great. I've made a change to make the XFormCaseBlock lazy: 3a1f087e2ac

Comment (orangejenny) [reply corehq/apps/app_manager/xform.py]: Thank you, it's easier to follow now.

Comment (snopoke) [reply corehq/apps/app_manager/xform.py]: seems that's broken a bunch of tests

Comment (snopoke) [reply corehq/apps/app_manager/xform.py:1576]: reverted that change since it seems it's more effort than it's worth

Comment (snopoke) [reply corehq/apps/app_manager/xform.py:1576]: I took another pass: https://github.com/dimagi/commcare-hq/pull/30532
